include/index_search_context.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "percentile_stats.h"
+#include "timer.h"
+#include "windows_customizations.h"
+
+namespace diskann
+{
+
+/// <summary>
+/// Search state
+/// </summary>
+enum State : uint8_t
+{
+    Unknown = 0,
+    Success = 1,
+    Failure = 2,
+    FailureTimeout = 3,      // Fail because of timeout
+    FailureException = 4,    // Fail because of Exception
+    FailureInvalidLabel = 5, // Fail because of invalid label
```

Comment:
IT seems FailureInvalidLabel is not used anywhere in the code at present. Is it necessary?